relayer/channel.go

```diff
@@ -22,881 +20,133 @@ func (c *Chain) CreateOpenChannels(
-				zap.String("src_chain_id", src.ChainID()),
-				zap.String("dst_chain_id", dst.ChainID()),
-			)
-		}
-
-		return existingChanID, dstChanID, true, true, nil
-
-	// OpenTry on counterparty
-	// source channel exists, but counterparty channel does not exist
-	case srcChanID != "" && dstChanID == "":
-		dst.log.Debug(
-			"Attempting to open channel end",
-			zap.String("src_chain_id", dst.ChainID()),
-			zap.String("dst_chain_id", src.ChainID()),
-		)
-
-		if !override {
-			existingChanID, found = FindMatchingChannel(ctx, dst, dstPortID, srcPortID, order, version)
-		}
-
-		if !found || override {
-
-			if err = retry.Do(func() error {
-				srch, err = src.ChainProvider.QueryLatestHeight(ctx)
-				if err != nil || srch == 0 {
-					return fmt.Errorf("failed to query latest heights. Err: %w", err)
-				}
-				return err
-			}, retry.Context(ctx), RtyAtt, RtyDel, RtyErr); err != nil {
-				return srcChanID, dstChanID, false, false, err
-			}
-
-			if err = retry.Do(func() error {
-				srcHeader, err = src.ChainProvider.GetIBCUpdateHeader(ctx, srch, dst.ChainProvider, dst.ClientID())
-				if err != nil || srch == 0 {
-					return fmt.Errorf("failed to get IBC update header. Err: %w", err)
-				}
-				return err
-			}, retry.Context(ctx), RtyAtt, RtyDel, RtyErr, retry.OnRetry(func(n uint, err error) {
-				src.LogRetryGetIBCUpdateHeader(n, err)
-				srch, _ = src.ChainProvider.QueryLatestHeight(ctx)
-			})); err != nil {
-				return srcChanID, dstChanID, false, false, err
-			}
-
-			// open try on destination chain
-			msgs, err = dst.ChainProvider.ChannelOpenTry(ctx, src.ChainProvider, srcHeader, dstPortID, srcPortID, dstChanID, srcChanID, version, dst.ConnectionID(), dst.ClientID())
-			if err != nil {
-				return srcChanID, dstChanID, false, false, err
-			}
-
-			res, success, err = dst.ChainProvider.SendMessages(ctx, msgs)
-			if err != nil {
-				dst.LogFailedTx(res, err, msgs)
-			}
-			if !success {
-				return srcChanID, dstChanID, false, false, err
-			}
-
-			// use index 1, channel open try is the second message in the transaction
-			existingChanID, err = ParseChannelIDFromEvents(res.Events)
-			if err != nil {
-				return srcChanID, dstChanID, false, false, err
-			}
-		} else {
-			dst.log.Debug(
-				"Channel end already exists",
-				zap.String("channel_id", existingChanID),
-				zap.String("src_chain_id", dst.ChainID()),
-				zap.String("dst_chain_id", src.ChainID()),
-			)
-		}
-
-		return srcChanID, existingChanID, true, true, nil
+	c.log.Info("Starting event processor for channel handshake",
+		zap.String("src_chain_id", c.PathEnd.ChainID),
+		zap.String("src_port_id", srcPortID),
+		zap.String("dst_chain_id", dst.PathEnd.ChainID),
+		zap.String("dst_port_id", dstPortID),
+	)
 
-	default:
-		return srcChanID, dstChanID, false, false, fmt.Errorf("channel ends already created")
-	}
+	return processor.NewEventProcessor().
+		WithChainProcessors(
+			srcPathChain.chainProcessor(c.log),
+			dstPathChain.chainProcessor(c.log)).
```

Comment:
The `).` can go to its own line for visual symmetry with the rest of the builder pattern here.

Reply:
done